src/core.jl

```diff
@@ -256,7 +256,8 @@ end
 function async_push!(n, x, onerror=print_error)
     taken = Base.n_avail(_messages)
     if taken >= CHANNEL_SIZE
-        warn("Message queue is full. Ordering may be incorrect.")
+        warn("Message queue is full. Ordering may be incorrect. "*
+        "Channel size can be increased by setting `ENV[\"REACTIVE_CHANNEL_SIZE\"] = ...`")
```

Comment:
Nice touch 😄 
maybe we should also add "by setting `ENV[\"REACTIVE_CHANNEL_SIZE\"] = ...`" before `using Reactive`" ?

Reply:
Hm..., that won't be enough because of pre-compilation freezing the value into the constant.

Reply:
oh shit, you're right... maybe we need to change `CHANNEL_SIZE` to be a Ref ?

Reply:
As in
```
const CHANNEL_SIZE = Ref(1024)
CHANNEL_SIZE[] = parse(Int, get(ENV, "REACTIVE_CHANNEL_SIZE", "1024"))
```
or something

Reply:
maybe `haskey(ENV, "REACTIVE_CHANNEL_SIZE") && (CHANNEL_SIZE[] = parse(Int, ENV["REACTIVE_CHANNEL_SIZE"])` is better (less repetition of 1024)

Reply:
Oh, that needs to go in an `__init__()`

Reply:
https://github.com/JuliaGizmos/Reactive.jl/blob/master/src/core.jl#L417

Reply:
What is the advantage of
```
const CHANNEL_SIZE = Ref(parse(Int, get(ENV, "REACTIVE_CHANNEL_SIZE", "1024")))
```
 over 
```
CHANNEL_SIZE = parse(Int, get(ENV, "REACTIVE_CHANNEL_SIZE", "1024"))
```
?

Reply:
Because then we can set the Ref in the `__init__()` function. And it will be updated based on the current value of `ENV["REACTIVE_CHANNEL_SIZE"]` on `using Reactive`

Reply:
I just tried out `const CHANNEL_SIZE = Ref(1024)`
and in `__init__()`

```
    haskey(ENV, "REACTIVE_CHANNEL_SIZE") && 
        (CHANNEL_SIZE[] = parse(Int, ENV["REACTIVE_CHANNEL_SIZE"]))
    runner_task[] = @async run()
```
plus changing lines ~226, ~257 from `CHANNEL_SIZE` to `CHANNEL_SIZE[]` and it seems to respect my `ENV["REACTIVE_CHANNEL_SIZE"]`

Reply:
Oh, I missed that your CHANNEL_SIZE is not `const`. The const is an advantage for static compilation, and because non-const globals can be slow.

Reply:
Ah, I get it, `const x = Ref` makes sure that can infer `Int` because only the value but not the type can change. https://github.com/JuliaLang/julia/issues/13817#issuecomment-152157853